syntax = "proto3";
package pt.ulisboa.tecnico.name.server.grpc;

message RegisterRequest{
    string serviceName = 1;
    string qualifier = 2;
    string address = 3;
}

message RegisterResponse{
}

message RegisterClientRequest{
}

message RegisterClientResponse{
    int32 id = 1;
}

message LookupRequest {
    string serviceName = 1;
    string qualifier = 2;
}

message LookupResponse {
    repeated string servers = 1;
}

message DeleteRequest {
    string serviceName = 1;
    string address = 2;
}

message DeleteResponse {
}

service NameServer{
    rpc register(RegisterRequest) returns (RegisterResponse);
    rpc registerClient(RegisterClientRequest) returns (RegisterClientResponse);
    rpc lookup(LookupRequest) returns (LookupResponse);
    rpc delete(DeleteRequest) returns (DeleteResponse);
}
